### PR TITLE
Use history replaceState instead of redirect for SSO flow

### PR DESCRIPTION
**Pull request by t3chguy**
Has additional benefit of not leaving the loginToken in history

### CODE REVIEW

**Review by jryans — APPROVED** [review]
Thanks, this seems sensible to me. It does mean the original query params may still be in bits of app state if they were captured on load, but it doesn't seem like that will cause problems or security risks as far as I can see.

**Comment by t3chguy** [comment]
The token is single use, purging it from history & url is just a cleanliness thing

**Comment by bekliev** [comment]
@t3chguy if another params are passed alongside `loginToken` - will they remain in window.location API?

File: src/vector/app.tsx [hunk 1]
```diff
@@ -124,7 +124,7 @@ function onTokenLoginCompleted() {
     parsedUrl.search = "";
```
Comment (bekliev) [inline src/vector/app.tsx:124]: Maybe it's better to just clean particular query param?
In our case it's the `loginToken`

**Comment by t3chguy** [comment]
@bekliev maybe, I was just matching the behaviour that was already there and swapping the redirect for a replaceState

**Pull request merged**